themes/classic/templates/catalog/_partials/product-prices.tpl

```diff
@@ -50,7 +50,7 @@
 
     {block name='product_unit_price'}
       {if $displayUnitPrice}
-        <p class="product-unit-price">{convertPrice price=$unit_price} {l s='per %s' sprintf=$product.unity}</p>
+        <p class="product-unit-price"{hook h='displayProductPriceBlock' product=$product type="unit_price" returntype="style"}>{convertPrice price=$unit_price} {l s='per %s' sprintf=$product.unity}</p>
```

Comment:
There should be no empty `<p>`
In product controller, we have this: `<p class="product-unit-price">{convertPrice price=$unit_price} {l s='per %s' sprintf=$product.unity}</p>`
And this is only shown, if $displayUnitPrice is true.